command/agent/exec/exec.go

```diff
@@ -333,3 +367,19 @@ func (s *Server) restartChildProcess(newEnvVars []string) error {
 
 	return nil
 }
+
+func (s *Server) Close() {
+	s.childProcessLock.Lock()
+	defer s.childProcessLock.Unlock()
+	s.close()
+}
+
+func (s *Server) close() {
+	if s.childProcessCloseStdOut {
```

Comment:
Out of curiosity -- why do we need this? I checked the documentation and it doesn't look like there'd be any issue calling `os.Stdout.Close()`, but I'd love to hear more!

Reply:
I'm honestly not sure... I know not to close stdout with python, but not sure with go

would have to look up the order of things, wouldn't want to close stdout then if something (like consul-template for example) tries to write to it

Reply:
It would be nice if we could avoid this additional complexity, so I think it would be nice if we could do it without the `childProcessCloseStdOut` yeah. From what I can tell, nothing seems to be wrong with closing stdout in go, and we'll be closing the process anyway, right? So there's no situation where we'd close it and another process will need it, right?

Reply:
correct! will try it (looks a lot simpler this way) thanks!